pydantic/_internal/_generate_schema.py

```diff
@@ -1106,6 +1116,72 @@ def _get_or_cache_recursive_ref(self, cls: type[Any]) -> tuple[str, core_schema.
             self.recursion_cache[obj_ref] = core_schema.definition_reference_schema(obj_ref)
             return obj_ref, None
 
+    def _annotated_schema(self, annotated_type: Any) -> core_schema.CoreSchema:
+        """
+        Generate schema for an Annotated type, e.g. `Annotated[int, Field(...)]` or `Annotated[int, Gt(0)]`.
+        """
+        first_arg, *other_args = get_args(annotated_type)
+        return self.apply_annotations(CallbackGetCoreSchemaHandler(self.generate_schema), first_arg, other_args)
+
+    def _get_prepare_pydantic_annotations_for_known_type(self, obj: Any) -> Callable[..., Any] | None:
```

Comment:
I'm a bit confused, is this temporary?

I'd rather we do this thing with decimals properly in this PR to demonstrate behaviour.

Reply:
We have a mix of places where we handle "known types". Some of it is in `GenerateSchema._generate_schema`, and some of it is in `_standard_types.py`. Decimal used to be in the latter, but the system in `_standard_types.py` doesn't really work with this new API. I could put thins is in `GenerateSchema._generate_schema` but that's already huge. So yes this is most likely temporary, although I think it may end up being the final resting place for Decimal and things like it.